README.md

```diff
@@ -4,7 +4,290 @@
 
-Describe pcic-react-external-text here.
+## Purpose
+
+This package enables an application to externalize part or all of its text content
+in what is known in some circles (e.g., Java, .Net) as a resource file.
+Instead of being literally included in the source of the application,
+text resources (from a one-word label to a phrase to an entire page of help)
+are referred to indirectly by an identifier (e.g., 'app.name'), and
+rendered using the ExternalText component. What is rendered depends on the
+content of the resource file. This decouples maintenance of the text
+from maintenance of the app that uses it, and also simplifies reusing the
+same text in different places in an app.
+
+For convenience, the content of an item in the resource file can be
+(and by default is) interpreted as Markdown and rendered into HTML.
+This is the most common usage.
+
+For other uses an item in the resource file can also be interpreted and rendered 
+as a simple string.
+
+Because texts in actual applications frequently need to include variable data from
+the app, external texts are treated as JavaScript template literals, and
+are evaluated in the context of an optional user-provided dictionary of
+data values. For example, an external text may be the string
+`You have ${num} messages.`. The app can provide a dictionary containing 
+a value for `num` which is interpolated into the rendered text.
+
+To further support modularization of texts, elements of the external text
+source itself can be referred to within an external text string, courtesy
+of the automatically provided context variable `$$`. For example, the text
+`This application is called ${$$.app.name}` includes content of the the
+item at path `app.name` in its rendering (e.g., the rendered text might
+be `This application is called Climate Explorer`). Such self-reference
+can be nested indefinitely deep.
+
+This module is not too far from an internationalization (i18n) package,
+but is considerably simpler and lighter-weight. It also provides
+Markdown interpretation (which admittedly could be wrapped around an i18n
+package).
+
+## Installation
+
+Depends on whether we actually publish to npm. If so:
+
+```
+npm install pcic-react-external-text
+```
+
+To install directly from GitHub:
+
+```
+npm install git+https://git@github.com/pacificclimate/pcic-react-external-text.git#<version>
+```
+
+## Demo
+
+To run the package demo:
+
+```shell script
+npm install
+npm start
+```
+
+After `npm start` should see:
+
+```text
+Compiled successfully in 2099 ms.
+
+The app is running at http://localhost:3000/
+```
+
+Browse to `localhost:3000` and you should see a React app that shows  page of text created via 
+`pcic-react-external-text`.
+
+## Usage
+
+1. Prepare an external texts file. (This is usually done in parallel with
+application development.) This file can be in any format, but ultimately
+must be converted to JS object for consumption by `ExternalText`.
```

Comment:
Minor issue: I found this phrasing confusing on the first readthrough. I parsed as "converted ... by ExternalText" instead of the intended "... consumption by ExternalText". 

It did become clear eventually, a few paragraphs in I realized I was confused and came back and reread this sentence.